### PR TITLE
extractbits_exprt has a single correct operand order

### PR DESCRIPTION
**Pull request by tautschnig**
Let's not trade some sort of "userfriendliness" for an unclear semantics.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by peterschrammel — APPROVED** [review]
:+1:

File: src/solvers/flattening/boolbv_extractbits.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ bvt boolbvt::convert_extractbits(const extractbits_exprt &expr)
     expr.find_source_location(),
     irep_pretty_diagnosticst{expr});
 
-  if(lower_as_int > upper_as_int)
-    std::swap(upper_as_int, lower_as_int);
+  DATA_INVARIANT(
+    o1 <= o2, "upper bound must be greater or equal to lower bound");
```
Comment (peterschrammel) [inline src/solvers/flattening/boolbv_extractbits.cpp]:
```suggestion
    lower_as_int <= upper_as_int, "upper bound must be greater or equal to lower bound");
```

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0e0deff32c8cbb9d04e8b117f19decc8253d9e0b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93278090

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4274761a5c264a45b6e4ba3bdd0d3ad72e120916).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93295549

**Pull request merged**